zeebe/stream-platform/src/main/java/io/camunda/zeebe/stream/impl/ProcessingStateMachine.java

```diff
@@ -361,6 +361,8 @@ private void batchProcessing(final TypedRecord<?> initialCommand) {
 
       final var command = pendingCommands.removeFirst();
 
+      processingResultBuilder.setRequestId(command.getRequestId());
```

Comment:
🔧 We set `requestId` here for each follow up command. Although, it should be enough if we set it only once from the initial command.

I couldn't validate my assumption. Would be good to verify with tests and make the change if it works.

Reply:
If you notice `TypedEventWriter.appendFollowUpEvent()` is called from places where mostly there is no access to the `requestId`, and then later in the stacktrace when the `ProcessingResultBuilder.append()` was called to append those events, the `metadata` was constructed without a `requestId` as well (since there is no access to get it).
Hence, I needed to rely on the `ProcessingResultBuilder` instantiated in the beginning of processing (in the `ProcessingStateMachine`) to keep the original requestId in its state so it can be appended to the followup requests.

I'm open to have a call about this to explore different possibilities if you like. It looks ugly, and like a non-intuitive workaround, but I tried to look into several ways around it, and so far I failed.

Reply:
Alright, I overlook that we built metadata for followup events and commands as a new object. I thought, we pass the metadata along from the initial command.